server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/ClusterBalanceStatsTests.java

```diff
@@ -300,18 +428,70 @@ private static ClusterState createClusterState(List<DiscoveryNode> nodes, List<T
             .build();
     }
 
-    private static DesiredBalance createDesiredBalance(ClusterState state) {
+    private static DesiredBalance createDesiredBalance(ClusterState state, double nodeWeight) {
+        Tuple<Map<ShardId, ShardAssignment>, Map<String, Long>> assignmentsAndShardCounts = computeAssignmentsAndShardCounts(state);
+        Map<ShardId, ShardAssignment> assignments = assignmentsAndShardCounts.v1();
+        Map<String, Long> shardCounts = assignmentsAndShardCounts.v2();
+
+        final Map<DiscoveryNode, DesiredBalanceMetrics.NodeWeightStats> nodeWeights = computeNodeWeights(state, shardCounts, nodeWeight);
+
+        return new DesiredBalance(1, assignments, nodeWeights, DesiredBalance.ComputationFinishReason.CONVERGED);
+    }
+
+    private static DesiredBalance createDesiredBalanceWithoutSpecificNode(RoutingNode nodeToRemove, ClusterState state, double nodeWeight) {
+        Tuple<Map<ShardId, ShardAssignment>, Map<String, Long>> assignmentsAndShardCounts = computeAssignmentsAndShardCounts(state);
+        Map<ShardId, ShardAssignment> assignments = assignmentsAndShardCounts.v1();
+        Map<String, Long> shardCounts = assignmentsAndShardCounts.v2();
+
+        final Map<DiscoveryNode, DesiredBalanceMetrics.NodeWeightStats> nodeWeights = computeNodeWeights(state, shardCounts, nodeWeight);
+
+        // Remove the node weights for the specified node
+        nodeWeights.remove(nodeToRemove.node());
+
+        return new DesiredBalance(1, assignments, nodeWeights, DesiredBalance.ComputationFinishReason.CONVERGED);
+    }
+
+    private static DesiredBalance createDesiredBalanceWithEmptyNodeWeights(ClusterState state) {
+        Tuple<Map<ShardId, ShardAssignment>, Map<String, Long>> assignmentsAndShardCounts = computeAssignmentsAndShardCounts(state);
+        Map<ShardId, ShardAssignment> assignments = assignmentsAndShardCounts.v1();
+
+        return new DesiredBalance(1, assignments, new HashMap<>(), DesiredBalance.ComputationFinishReason.CONVERGED);
```

Comment:
Nit: no need to instantiate an actual map
```suggestion
        return new DesiredBalance(1, assignments, Map.of(), DesiredBalance.ComputationFinishReason.CONVERGED);
```